tests/testthat/_snaps/conditions.md

```diff
@@ -14,3 +14,30 @@
       Error:
       ! 1
 
+# errors have useful call
+
+    Code
+      cnd$trace
+    Output
+           x
+        1. +-evaluate (local) `<fn>`()
+        2. | +-base::withRestarts(...) at evaluate/R/evaluate.R:144:7
+        3. | | \-base (local) withRestartList(expr, restarts)
+        4. | |   +-base (local) withOneRestart(withRestartList(expr, restarts[-nr]), restarts[[nr]])
+        5. | |   | \-base (local) doWithOneRestart(return(expr), restart)
+        6. | |   \-base (local) withRestartList(expr, restarts[-nr])
+        7. | |     +-base (local) withOneRestart(withRestartList(expr, restarts[-nr]), restarts[[nr]])
+        8. | |     | \-base (local) doWithOneRestart(return(expr), restart)
+        9. | |     \-base (local) withRestartList(expr, restarts[-nr])
+       10. | |       \-base (local) withOneRestart(expr, restarts[[1L]])
+       11. | |         \-base (local) doWithOneRestart(return(expr), restart)
+       12. | +-evaluate:::with_handlers(...) at evaluate/R/evaluate.R:144:7
+       13. | | +-base::eval(call) at evaluate/R/conditions.R:50:3
+       14. | | | \-base::eval(call)
+       15. | | \-base::withCallingHandlers(...)
+       16. | \-base::withVisible(do) at evaluate/R/evaluate.R:153:15
+       17. \-evaluate (local) f() at evaluate/R/evaluate.R:153:15
```

Comment:
At least we now get the actual error traceback. Is there some obvious way to trim off some of the earlier frames?

Reply:
We can no longer use frame environments as boundaries (or `top_env`) because we no longer evaluate in an `eval()` stack frame. We might have to introduce a new way of setting a boundary in the call stack...

Reply:
Hmmm that doesn't make any difference.

Reply:
Did you set the `rlang_trace_top_env` option (from memory) in that wrapper?

Reply:
I tried this in both `evaluate()` and `evaluate_()`:

```R
old <- options(rlang_trace_top_env = environment())
defer(options(old))
```

Reply:
I also wonder if this is the root cause of #219